test/cmdlineTests/model_checker_print_smtlib_code_all/err

```diff
@@ -0,0 +1,234 @@
+Info: CHC: SMTLIB code used for formal verification:
+(declare-datatypes ((state_type 0)) (((state_type (balances (Array Int Int))))))
+(declare-datatypes ((bytes_tuple 0)) (((bytes_tuple (bytes_tuple_accessor_array (Array Int Int)) (bytes_tuple_accessor_length Int)))))
+(declare-datatypes ((ecrecover_input_type 0)) (((ecrecover_input_type (hash Int) (v Int) (r Int) (s Int)))))
+(declare-datatypes ((crypto_type 0)) (((crypto_type (ecrecover (Array ecrecover_input_type Int)) (keccak256 (Array bytes_tuple Int)) (ripemd160 (Array bytes_tuple Int)) (sha256 (Array bytes_tuple Int))))))
+(declare-datatypes ((abi_type 0)) (((abi_type))))
+(declare-datatypes ((tx_type 0)) (((tx_type (block.basefee Int) (block.chainid Int) (block.coinbase Int) (block.gaslimit Int) (block.number Int) (block.prevrandao Int) (block.timestamp Int) (blockhash (Array Int Int)) (msg.data bytes_tuple) (msg.sender Int) (msg.sig Int) (msg.value Int) (tx.gasprice Int) (tx.origin Int)))))
+(declare-rel block_8_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type Int))
+(declare-rel summary_constructor_2_C_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel block_9_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type Int))
+(declare-rel summary_4_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel contract_initializer_after_init_12_C_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel error_target_3_0 ())
+(declare-rel contract_initializer_entry_11_C_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel block_5_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type Int))
+(declare-rel nondet_interface_1_C_16_0 (Int Int abi_type crypto_type state_type state_type))
+(declare-rel block_6_f_14_16_0 (Int Int abi_type crypto_type tx_type state_type state_type Int))
+(declare-rel contract_initializer_10_C_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel block_7_return_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type Int))
+(declare-rel implicit_constructor_entry_13_C_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-rel interface_0_C_16_0 (Int abi_type crypto_type state_type))
+(declare-rel summary_3_function_f__15_16_0 (Int Int abi_type crypto_type tx_type state_type state_type))
+(declare-var A tx_type)
+(declare-var B Int)
+(declare-var C state_type)
+(declare-var D Int)
+(declare-var E crypto_type)
+(declare-var F abi_type)
+(declare-var G state_type)
+(declare-var H state_type)
+(declare-var I Int)
+(declare-var J Int)
+(declare-var K Int)
+(declare-var L Int)
+(declare-var M Bool)
+(declare-var N Int)
+(declare-var O Int)
+(declare-var P Int)
+(declare-var Q state_type)
+(declare-var R Int)
+(rule (! (=> (= D 0) (nondet_interface_1_C_16_0 D B F E C C)) :named base_nondet))
```

Comment:
Yea the annoying thing of dumping from z3 directly is that it uses its own `rule` language which can be seen here, and is not part of the smtlib standard. Maybe we should require users to use the smtlib2 solver if they want to see queries.

Reply:
Good idea.